components/jupyter-web-app/backend/kubeflow_jupyter/common/api.py

```diff
@@ -111,149 +77,95 @@ def wrap(fn, *args, **kwargs):
 
 # API Functions
 # GETers
-def get_pvcs(ns, user=None):
-    if not is_authorized(user, ns):
-        return {
-            "success": False,
-            "log": "User '{}' is not authorized for namespace '{}'".format(
-                user, ns
-            )
-        }
-
+@auth.needs_authorization("list", "", "v1", "persistentvolumeclaims")
+def list_pvcs(namespace):
     return wrap_resp(
         "pvcs",
-        v1_core.list_namespaced_persistent_volume_claim, namespace=ns
+        v1_core.list_namespaced_persistent_volume_claim,
+        namespace=namespace
     )
 
 
-def get_pods(ns, user=None):
-    if not is_authorized(user, ns):
-        return {
-            "success": False,
-            "log": "User '{}' is not authorized for namespace '{}'".format(
-                user, ns
-            )
-        }
-
-    return wrap_resp(
-        "pods",
-        v1_core.list_namespaced_pod, namespace=ns
-    )
-
-
-def get_notebooks(ns, user=None):
-    if not is_authorized(user, ns):
-        return {
-            "success": False,
-            "log": "User '{}' is not authorized for namespace '{}'".format(
-                user, ns
-            )
-        }
-
+@auth.needs_authorization("list", "kubeflow.org", "v1beta1", "notebooks")
+def list_notebooks(namespace):
     return wrap_resp(
         "notebooks",
         custom_api.list_namespaced_custom_object,
         "kubeflow.org",
         "v1beta1",
-        ns,
+        namespace,
         "notebooks"
     )
 
 
-def get_poddefaults(ns, user=None):
-    if not is_authorized(user, ns):
-        return {
-            "success": False,
-            "log": "User '{}' is not authorized for namespace '{}'".format(
-                user, ns
-            )
-        }
-
+@auth.needs_authorization("list", "kubeflow.org", "v1alpha1", "poddefaults")
+def list_poddefaults(namespace):
     return wrap_resp(
         "poddefaults",
         custom_api.list_namespaced_custom_object,
         "kubeflow.org",
         "v1alpha1",
-        ns,
+        namespace,
         "poddefaults"
     )
 
 
-def get_namespaces(user=None):
+@auth.needs_authorization("get", "core", "v1", "secrets")
+def get_secret(name, namespace):
     return wrap_resp(
-        "namespaces",
-        v1_core.list_namespace
+        "secret",
+        v1_core.read_namespaced_secret,
+        name,
+        namespace
     )
 
 
-def get_storageclasses(user=None):
+@auth.needs_authorization("list", "core", "v1", "namespaces")
+def list_namespaces():
     return wrap_resp(
-        "storageclasses",
-        storage_api.list_storage_class
+        "namespaces",
+        v1_core.list_namespace
     )
 
 
-def get_secret(ns, nm, user=None):
-    if not is_authorized(user, ns):
-        return {
-            "success": False,
-            "log": "User '{}' is not authorized for namespace '{}'".format(
-                user, ns
-            )
-        }
-
+# @auth.needs_authorization("list", "storage.k8s.io", "v1", "storageclasses")
```

Comment:
uncomment?

Reply:
Currently users added with the Profile flow (either owner or edit/view) will get a `RoleBinding` that will give them `kubeflow-admin` permissions on this namespace.

But, in order to list StorageClasses the user will need to have a `ClusterRoleBinding` which is not currently supported.  To keep the current UX I let JWA to use its own credentials for this call. I will open an issue for us to keep track of this detail and discuss whether it makes sense to change the Profiles mechanism to support this case.

Also, as a side note, JWA exposes only one endpoint `/api/storageclasses/default` which will use this function in order to determine if a default storageclass is set. So currently an unauthorized user will not be able to use this function to list storageclasses.